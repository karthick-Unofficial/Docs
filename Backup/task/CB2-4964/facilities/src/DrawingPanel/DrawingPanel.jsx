import React from "react";
import PropTypes from "prop-types";
import { Drawer } from "@material-ui/core";
import FacilityFormContainer from "./FacilityForm/FacilityFormContainer";
import FloorPlanFormContainer from "./FloorPlanForm/FloorPlanFormContainer";
import CameraFormContainer from "./CameraForm/CameraFormContainer";
import AccessPointFormContainer from "./AccessPointForm/AccessPointFormContainer";

const propTypes = {
	mode: PropTypes.string,
	type: PropTypes.string,
	dir: PropTypes.string
};

const defaultProps = {
	mode: null,
	type: null
};

const DrawingPanel = ({ mode, type, dir }) => {
	return (
		<Drawer
			open={!!mode}
			anchor={dir === "ltr" ? "left" : "right"}
			variant="persistent"
			PaperProps={{
				style: {
					width: 350,
					height: "calc(100vh - 48px)",
					top: 48
				}
			}}
		>
			<div style={{ padding: 16 }}>
				{type === "facility" && <FacilityFormContainer />}
				{type === "floor-plan" && <FloorPlanFormContainer />}
				{type === "camera" && <CameraFormContainer />}
				{type === "access point" && <AccessPointFormContainer/>}
			</div>
		</Drawer>
	);
};

DrawingPanel.propTypes = propTypes;
DrawingPanel.defaultProps = defaultProps;

export default DrawingPanel;
