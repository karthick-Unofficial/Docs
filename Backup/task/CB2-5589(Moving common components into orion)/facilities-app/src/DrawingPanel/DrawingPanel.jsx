import React from "react";
import { Drawer } from "@mui/material";
import FacilityForm from "./FacilityForm/FacilityForm";
import CameraForm from "./CameraForm/CameraForm";
import AccessPointForm from "./AccessPointForm/AccessPointForm";
import FloorPlanForm from "./FloorPlanForm/FloorPlanForm";
import { useSelector } from "react-redux";
import { getDir } from "orion-components/i18n/Config/selector";


const DrawingPanel = () => {

	const { mode, type } = useSelector(state => state.mapState.mapTools);
	const dir = useSelector(state => getDir(state));

	return (
		<Drawer
			open={!!mode}
			anchor={dir === "ltr" ? "left" : "right"}
			variant="persistent"
			PaperProps={{
				style: {
					width: 350,
					height: "calc(100vh - 48px)",
					top: 48
				}
			}}
		>
			<div style={{ padding: 16 }}>
				{type === "facility" && <FacilityForm />}
				{type === "floor-plan" && <FloorPlanForm />}
				{type === "camera" && <CameraForm />}
				{type === "accessPoint" && <AccessPointForm />}
			</div>
		</Drawer>
	);
};

export default DrawingPanel;
