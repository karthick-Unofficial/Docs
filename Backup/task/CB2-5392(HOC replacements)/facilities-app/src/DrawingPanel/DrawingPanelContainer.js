/*import { connect } from "react-redux";
import DrawingPanel from "./DrawingPanel.jsx";
import { getDir } from "orion-components/i18n/Config/selector";

const mapStateToProps = state => {
	const { mode, type } = state.mapState.mapTools;
	return { mode, type, dir: getDir(state) };
};

const DrawingPanelContainer = connect(mapStateToProps)(DrawingPanel);

export default DrawingPanelContainer;
*/