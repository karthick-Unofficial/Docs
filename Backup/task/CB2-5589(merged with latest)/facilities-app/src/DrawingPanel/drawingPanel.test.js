import React from "react";
import DrawingPanel from "./DrawingPanel";
import { shallow } from "enzyme";
import { useSelector } from "react-redux";

jest.mock("react-redux", () => ({
	useSelector: jest.fn()
}));

describe("DrawingPanel", () => {
	it("renders", () => {
		useSelector.mockImplementation(() => {
			return { mapTools: { mode: {}, type: {} } };
		});
		const wrapper = shallow(<DrawingPanel />);
		expect(wrapper).toMatchSnapshot();
	});
});
