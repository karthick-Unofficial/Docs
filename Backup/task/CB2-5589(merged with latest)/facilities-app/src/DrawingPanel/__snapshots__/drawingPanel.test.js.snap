// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawingPanel renders 1`] = `ShallowWrapper {}`;
